<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.43">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="Quantifying chromatin accessibility">

<title>5&nbsp; ATAC-Seq â€“ SeqHub</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../chapters/meta/index.html" rel="next">
<link href="../../chapters/airr/index.html" rel="prev">
<link href="../../assets/favicon.png" rel="icon" type="image/png">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-39489dca9ddd9247a1c53ef4381789db.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-3d9ea01a5de0f98e0776051558484512.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/quarto-contrib/fontawesome6-0.1.0/all.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/fontawesome6-0.1.0/latex-fontsize.css" rel="stylesheet">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<style>html{ scroll-behavior: smooth; }</style>
<style>

      .quarto-title-block .quarto-title-banner h1,
      .quarto-title-block .quarto-title-banner h2,
      .quarto-title-block .quarto-title-banner h3,
      .quarto-title-block .quarto-title-banner h4,
      .quarto-title-block .quarto-title-banner h5,
      .quarto-title-block .quarto-title-banner h6
      {
        color: #f2f3f4;
      }

      .quarto-title-block .quarto-title-banner {
        color: #f2f3f4;
      }
</style>
<link rel="preconnect" href="https://fonts.googleapis.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="">
<link href="https://fonts.googleapis.com/css2?family=JetBrains+Mono&amp;family=Nunito:ital,wght@0,400;0,600;1,400;1,600&amp;display=swap" rel="stylesheet">


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <h1 class="quarto-secondary-nav-title no-breadcrumbs"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">ATAC-Seq</span></h1>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title d-none d-lg-block"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">ATAC-Seq</span></h1>
                  <div>
        <div class="description">
          Quantifying chromatin accessibility
        </div>
      </div>
                </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">SeqHub</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/royfrancis/seqhub" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text"><strong>Bulk RNASeq</strong></span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/bulk/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Bulk gene expression</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text"><strong>Single Cell RNASeq</strong></span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/sc/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Gene expression</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/spatial/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Spatial</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/airr/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">AIRR-Seq</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/atacseq/index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">ATAC-Seq</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text"><strong>Meta Transcriptomics</strong></span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../chapters/meta/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Meta transcriptomics</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#atacseq-overview" id="toc-atacseq-overview" class="nav-link active" data-scroll-target="#atacseq-overview"><span class="header-section-number">5.1</span> Overview</a>
  <ul class="collapse">
  <li><a href="#bulk-atac-seq" id="toc-bulk-atac-seq" class="nav-link" data-scroll-target="#bulk-atac-seq"><span class="header-section-number">5.1.1</span> Bulk ATAC-Seq</a></li>
  <li><a href="#single-cell-atac-seq" id="toc-single-cell-atac-seq" class="nav-link" data-scroll-target="#single-cell-atac-seq"><span class="header-section-number">5.1.2</span> Single-cell ATAC-Seq</a></li>
  </ul></li>
  <li><a href="#feature-selection" id="toc-feature-selection" class="nav-link" data-scroll-target="#feature-selection"><span class="header-section-number">5.2</span> Feature selection</a></li>
  <li><a href="#celltyping" id="toc-celltyping" class="nav-link" data-scroll-target="#celltyping"><span class="header-section-number">5.3</span> Celltyping</a></li>
  <li><a href="#differentially-accessible-regions" id="toc-differentially-accessible-regions" class="nav-link" data-scroll-target="#differentially-accessible-regions"><span class="header-section-number">5.4</span> Differentially accessible regions</a></li>
  <li><a href="#benchmarking-tools" id="toc-benchmarking-tools" class="nav-link" data-scroll-target="#benchmarking-tools"><span class="header-section-number">5.5</span> Benchmarking tools</a></li>
  <li><a href="#tools" id="toc-tools" class="nav-link" data-scroll-target="#tools"><span class="header-section-number">5.6</span> Tools</a></li>
  <li><a href="#tutorials" id="toc-tutorials" class="nav-link" data-scroll-target="#tutorials"><span class="header-section-number">5.7</span> Tutorials</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<section id="atacseq-overview" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="atacseq-overview"><span class="header-section-number">5.1</span> Overview</h2>
<section id="bulk-atac-seq" class="level3" data-number="5.1.1">
<h3 data-number="5.1.1" class="anchored" data-anchor-id="bulk-atac-seq"><span class="header-section-number">5.1.1</span> Bulk ATAC-Seq</h3>
<p><a href="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41596-022-00692-9/MediaObjects/41596_2022_692_Fig1_HTML.png?as=webp" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41596-022-00692-9/MediaObjects/41596_2022_692_Fig1_HTML.png?as=webp" class="img-fluid"></a></p>
<p>ATAC-seq provides a simple and scalable way to assay the regions of the genome that are bound by TFs, and to compare how these landscapes change between particular contexts or perturbations. This is accomplished using in vitro transposition of sequencing adapters into native chromatin. Each unique transposition event, termed an â€˜insertionâ€™, marks a location in the genome where a Tn5 transposase dimer is able to access DNA and perform a cut-and-paste reaction. The transposase simultaneously fragments the DNA and inserts sequence handles that are then used for amplification during library preparation.</p>
<p><span class="citation" data-cites="grandi2022chromatin">Grandi et al. (<a href="#ref-grandi2022chromatin" role="doc-biblioref">2022</a>)</span>, <span class="citation" data-cites="yan2020reads">Yan et al. (<a href="#ref-yan2020reads" role="doc-biblioref">2020</a>)</span></p>
</section>
<section id="single-cell-atac-seq" class="level3" data-number="5.1.2">
<h3 data-number="5.1.2" class="anchored" data-anchor-id="single-cell-atac-seq"><span class="header-section-number">5.1.2</span> Single-cell ATAC-Seq</h3>
<ul>
<li>ATAC-Seq quantifies DNA and can be applied to frozen/fixed tisues where nuclei can be isolated</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="https://media.springernature.com/full/springer-static/image/art%3A10.1007%2Fs42994-022-00082-5/MediaObjects/42994_2022_82_Fig1_HTML.png?as=webp" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Wet-lab workflow"><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1007%2Fs42994-022-00082-5/MediaObjects/42994_2022_82_Fig1_HTML.png?as=webp" class="img-fluid figure-img" alt="Wet-lab workflow"></a></p>
<figcaption>Wet-lab workflow</figcaption>
</figure>
</div>
<ul>
<li>Sample-level quality
<ul>
<li>Cell viability must exceed 80%</li>
<li>Accurate assessment of cell number</li>
</ul></li>
<li>Library-level quality
<ul>
<li>DNA fragment distribution (multiples of 200bp)</li>
</ul></li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="https://media.springernature.com/full/springer-static/image/art%3A10.1007%2Fs42994-022-00082-5/MediaObjects/42994_2022_82_Fig2_HTML.png?as=webp" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Data analysis workflow"><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1007%2Fs42994-022-00082-5/MediaObjects/42994_2022_82_Fig2_HTML.png?as=webp" class="img-fluid figure-img" alt="Data analysis workflow"></a></p>
<figcaption>Data analysis workflow</figcaption>
</figure>
</div>
<ul>
<li>Counts QC
<ul>
<li>Unique nuclear fragments (&gt;1000)</li>
<li>Fraction of transposition events in the peaks (&gt;0.3)</li>
<li>Transcription start sites (TSS) enrichment score (&gt;5)</li>
<li>Ratio of mononucleosomal to nucleosome-free fragments</li>
<li>Apply filters separately for each sample</li>
</ul></li>
<li>Generate features (peaks/target regions)
<ul>
<li>Various methods exist</li>
</ul></li>
<li>Dimensionality reduction</li>
<li>Cells with similar accessibility profiles are organized into clusters</li>
<li>There are two approaches to cell identity annotation: the cell type-specific peaks-based method and the scRNA-seq-based method
<ul>
<li>Enhancers can be used to accurately annotate cell types as distal cis-regulatory elements specific to particular cell types and states.</li>
<li>Cell type-specific gene expression is predicted based on their accessibility and used to annotate cells</li>
</ul></li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="https://media.springernature.com/full/springer-static/image/art%3A10.1007%2Fs42994-022-00082-5/MediaObjects/42994_2022_82_Fig3_HTML.png?as=webp" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Epigenomic profiling from scATAC-seq data."><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1007%2Fs42994-022-00082-5/MediaObjects/42994_2022_82_Fig3_HTML.png?as=webp" class="img-fluid figure-img" alt="Epigenomic profiling from scATAC-seq data."></a></p>
<figcaption>Epigenomic profiling from scATAC-seq data.</figcaption>
</figure>
</div>
<ul>
<li>Cell type-specific chromatin architecture
<ul>
<li>Profiling the regulatory elements for each cluster/cell type</li>
<li>Identifying differentially accessible regions between different clusters/cell types</li>
<li>Uncovering key factors that contribute to the altered chromatin accessibility</li>
<li>Linking promoterâ€“enhancer interactions</li>
</ul></li>
<li>Three main strategies are used to identify TFs of interest
<ul>
<li>Searching for overrepresented motifs in cell type-specific accessible regions</li>
<li>Comparing motif activity between cell types</li>
<li>Detecting foot-printing for TF occupancy</li>
</ul></li>
</ul>
<p><span class="citation" data-cites="shi2022fundamental">Shi et al. (<a href="#ref-shi2022fundamental" role="doc-biblioref">2022</a>)</span></p>
<p>Single-cell ATAC sequencing analysis: From data preprocessing to hypothesis generation</p>
<p><span class="citation" data-cites="baek2020single">Baek &amp; Lee (<a href="#ref-baek2020single" role="doc-biblioref">2020</a>)</span></p>
</section>
</section>
<section id="feature-selection" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="feature-selection"><span class="header-section-number">5.2</span> Feature selection</h2>
<p>The performance of various methods for analyzing datasets with different cell structures and sizes is discussed. For simple datasets with distinct cell types, all methods are effective. For datasets with small cell classes or with hierarchical clustering and similar subtypes, SnapATAC and SnapATAC2 are preferred. SnapATAC is not memory-efficient for large datasets (over 20,000 cells); in such cases, SnapATAC2 is better. Signac outperforms ArchR, but ArchR is more memory-efficient. Adding aggregation steps to Signac does not significantly increase time or memory usage. Feature engineering choices like peak versus bins calling do not majorly affect performance, so users can choose based on preference. Recommended latent space dimensions vary by method: 10-30 for SnapATAC/SnapATAC2, 10-50 for Signac/ArchR, and even larger for aggregation methods.</p>
<p><span class="citation" data-cites="de2024systematic">De Rop et al. (<a href="#ref-de2024systematic" role="doc-biblioref">2024</a>)</span></p>
</section>
<section id="celltyping" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="celltyping"><span class="header-section-number">5.3</span> Celltyping</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="assets/wang-2022-1.webp" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Performance of label transfer methods on single-cell data from selected mouse and human tissues. (A) Overall metrics considering performance on all scATAC-seq cells."><img src="assets/wang-2022-1.webp" class="img-fluid figure-img" alt="Performance of label transfer methods on single-cell data from selected mouse and human tissues. (A) Overall metrics considering performance on all scATAC-seq cells."></a></p>
<figcaption>Performance of label transfer methods on single-cell data from selected mouse and human tissues. (A) Overall metrics considering performance on all scATAC-seq cells.</figcaption>
</figure>
</div>
<p>Here, we evaluated the performance of five scATAC-seq annotation methods on both their classification accuracy and scalability using publicly available single-cell datasets from mouse and human tissues including brain, lung, kidney, PBMC, and BMMC. Using the BMMC data as basis, we further investigated the performance of these methods across different data sizes, mislabeling rates, sequencing depths and the number of cell types unique to scATAC-seq. Bridge integration, which is the only method that requires additional multimodal data and does not need gene activity calculation, was overall the best method and robust to changes in data size, mislabeling rate and sequencing depth. Conos was the most time and memory efficient method but performed the worst in terms of prediction accuracy. scJoint tended to assign cells to similar cell types and performed relatively poorly for complex datasets with deep annotations but performed better for datasets only with major label annotations. The performance of scGCN and Seurat v3 was moderate, but scGCN was the most time-consuming method and had the most similar performance to random classifiers for cell types unique to scATAC-seq.</p>
<p><span class="citation" data-cites="wang2022benchmarking">Wang et al. (<a href="#ref-wang2022benchmarking" role="doc-biblioref">2022</a>)</span></p>
</section>
<section id="differentially-accessible-regions" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="differentially-accessible-regions"><span class="header-section-number">5.4</span> Differentially accessible regions</h2>
<p><span class="citation" data-cites="gontarz2020comparison">Gontarz et al. (<a href="#ref-gontarz2020comparison" role="doc-biblioref">2020</a>)</span></p>
<ul>
<li>Best practices for scATAC-Seq
<ul>
<li>Wilcoxon rank-sum test was the most widely used for DA</li>
<li>Statistical methods that aggregated individual cells to form â€˜pseudobulksâ€™ generally yielded better DA results mainly due to reduced false discovery rates</li>
<li>Single-cell focused DA analysis methods often produced false discoveries, especially noticeable in data without biological differences</li>
</ul></li>
</ul>
<p><span class="citation" data-cites="teo2024best">Teo et al. (<a href="#ref-teo2024best" role="doc-biblioref">2024</a>)</span></p>
</section>
<section id="benchmarking-tools" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="benchmarking-tools"><span class="header-section-number">5.5</span> Benchmarking tools</h2>
<p>Our benchmarking results highlight SnapATAC, cisTopic, and Cusanovich2018 as the top performing scATAC-seq data analysis methods to perform clustering across all datasets and different metrics. Methods that preserve information at the peak level (cisTopic, Cusanovich2018, Scasat) or bin level (SnapATAC) generally outperform those that summarize accessible chromatin regions at the motif/k-mer level (chromVAR, BROCKMAN, SCRAT) or over the gene body (Cicero, Gene Scoring). In addition, methods that implement a dimensionality reduction step (BROCKMAN, cisTopic, Cusanovich2018, Scasat, SnapATAC) generally show advantages over the other methods without this important step. SnapATAC is the most scalable method; it was the only method capable of processing more than 80,000 cells. Cusanovich2018 is the method that best balances analysis performance and running time.</p>
<p><span class="citation" data-cites="chen2019assessment">Chen et al. (<a href="#ref-chen2019assessment" role="doc-biblioref">2019</a>)</span></p>
<p>Overall, feature aggregation, SnapATAC, and SnapATAC2 outperform latent semantic indexing-based methods. For datasets with complex cell-type structures, SnapATAC and SnapATAC2 are preferred. With large datasets, SnapATAC2 and ArchR are most scalable.</p>
<p><span class="citation" data-cites="luo2024benchmarking">Luo et al. (<a href="#ref-luo2024benchmarking" role="doc-biblioref">2024</a>)</span></p>
</section>
<section id="tools" class="level2" data-number="5.6">
<h2 data-number="5.6" class="anchored" data-anchor-id="tools"><span class="header-section-number">5.6</span> Tools</h2>
<p><img src="https://media.springernature.com/full/springer-static/esm/art%3A10.1038%2Fs41588-021-00790-6/MediaObjects/41588_2021_790_Fig5_ESM.jpg" class="img-fluid" alt="Comparison of toolkits"> <span class="citation" data-cites="granja2021archr">Granja et al. (<a href="#ref-granja2021archr" role="doc-biblioref">2021</a>)</span></p>
<ul>
<li><a href="https://stuartlab.org/signac/index.html"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> Signac</a> (R) <span class="citation" data-cites="stuart2021single">Stuart et al. (<a href="#ref-stuart2021single" role="doc-biblioref">2021</a>)</span></li>
<li><a href="https://github.com/GreenleafLab/ArchR"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> ArchR</a> (R) <span class="citation" data-cites="granja2021archr">Granja et al. (<a href="#ref-granja2021archr" role="doc-biblioref">2021</a>)</span></li>
<li><a href="https://github.com/r3fang/SnapATAC"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> SnapATAC</a> (R) <span class="citation" data-cites="fang2021comprehensive">Fang et al. (<a href="#ref-fang2021comprehensive" role="doc-biblioref">2021</a>)</span></li>
<li><a href="https://github.com/aertslab/pycisTopic"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> pycisTopic</a> (Python)</li>
<li><a href="https://github.com/ManchesterBioinference/Scasat"><i class="fa-solid fa-toolbox" aria-label="toolbox"></i> Scasat</a> (Bash,Python)</li>
</ul>
</section>
<section id="tutorials" class="level2" data-number="5.7">
<h2 data-number="5.7" class="anchored" data-anchor-id="tutorials"><span class="header-section-number">5.7</span> Tutorials</h2>
<ul>
<li><a href="https://www.youtube.com/watch?v=e2396GKFMRY"><i class="fa-brands fa-youtube" aria-label="youtube"></i> scATAC-Seq analysis in R using Signac by Sanbomics</a></li>
<li><a href="https://www.youtube.com/watch?v=yEKZJVjc5DY"><i class="fa-brands fa-youtube" aria-label="youtube"></i> scATAC-Seq analysis in R using Signac by Bioinformagician</a></li>
</ul>
</section>
<section id="references" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="references">References</h2>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-baek2020single" class="csl-entry" role="listitem">
Baek, S., &amp; Lee, I. (2020). Single-cell ATAC sequencing analysis: From data preprocessing to hypothesis generation. <em>Computational and Structural Biotechnology Journal</em>, <em>18</em>, 1429â€“1439. <a href="https://www.sciencedirect.com/science/article/pii/S2001037020303019">https://www.sciencedirect.com/science/article/pii/S2001037020303019</a>
</div>
<div id="ref-chen2019assessment" class="csl-entry" role="listitem">
Chen, H., Lareau, C., Andreani, T., Vinyard, M. E., Garcia, S. P., Clement, K., Andrade-Navarro, M. A., Buenrostro, J. D., &amp; Pinello, L. (2019). Assessment of computational methods for the analysis of single-cell ATAC-seq data. <em>Genome Biology</em>, <em>20</em>(1), 1â€“25. <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1854-5">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1854-5</a>
</div>
<div id="ref-de2024systematic" class="csl-entry" role="listitem">
De Rop, F. V., Hulselmans, G., Flerin, C., Soler-Vila, P., Rafels, A., Christiaens, V., GonzÃ¡lez-Blas, C. B., Marchese, D., Caratu, G., Poovathingal, S., et al. (2024). Systematic benchmarking of single-cell ATAC-sequencing protocols. <em>Nature Biotechnology</em>, <em>42</em>(6), 916â€“926. <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-024-03356-x">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-024-03356-x</a>
</div>
<div id="ref-fang2021comprehensive" class="csl-entry" role="listitem">
Fang, R., Preissl, S., Li, Y., Hou, X., Lucero, J., Wang, X., Motamedi, A., Shiau, A. K., Zhou, X., Xie, F., et al. (2021). Comprehensive analysis of single cell ATAC-seq data with SnapATAC. <em>Nature Communications</em>, <em>12</em>(1), 1337. <a href="https://www.nature.com/articles/s41467-021-21583-9">https://www.nature.com/articles/s41467-021-21583-9</a>
</div>
<div id="ref-gontarz2020comparison" class="csl-entry" role="listitem">
Gontarz, P., Fu, S., Xing, X., Liu, S., Miao, B., Bazylianska, V., Sharma, A., Madden, P., Cates, K., Yoo, A., et al. (2020). Comparison of differential accessibility analysis strategies for ATAC-seq data. <em>Scientific Reports</em>, <em>10</em>(1), 10150. <a href="https://www.nature.com/articles/s41598-020-66998-4">https://www.nature.com/articles/s41598-020-66998-4</a>
</div>
<div id="ref-grandi2022chromatin" class="csl-entry" role="listitem">
Grandi, F. C., Modi, H., Kampman, L., &amp; Corces, M. R. (2022). Chromatin accessibility profiling by ATAC-seq. <em>Nature Protocols</em>, <em>17</em>(6), 1518â€“1552. <a href="https://www.nature.com/articles/s41596-022-00692-9">https://www.nature.com/articles/s41596-022-00692-9</a>
</div>
<div id="ref-granja2021archr" class="csl-entry" role="listitem">
Granja, J. M., Corces, M. R., Pierce, S. E., Bagdatli, S. T., Choudhry, H., Chang, H. Y., &amp; Greenleaf, W. J. (2021). ArchR is a scalable software package for integrative single-cell chromatin accessibility analysis. <em>Nature Genetics</em>, <em>53</em>(3), 403â€“411. <a href="https://www.nature.com/articles/s41588-021-00790-6">https://www.nature.com/articles/s41588-021-00790-6</a>
</div>
<div id="ref-luo2024benchmarking" class="csl-entry" role="listitem">
Luo, S., Germain, P.-L., Robinson, M. D., &amp; Meyenn, F. von. (2024). Benchmarking computational methods for single-cell chromatin data analysis. <em>Genome Biology</em>, <em>25</em>(1), 225. <a href="https://link.springer.com/article/10.1186/s13059-024-03356-x">https://link.springer.com/article/10.1186/s13059-024-03356-x</a>
</div>
<div id="ref-shi2022fundamental" class="csl-entry" role="listitem">
Shi, P., Nie, Y., Yang, J., Zhang, W., Tang, Z., &amp; Xu, J. (2022). Fundamental and practical approaches for single-cell ATAC-seq analysis. <em>Abiotech</em>, <em>3</em>(3), 212â€“223. <a href="https://link.springer.com/article/10.1007/s42994-022-00082-5">https://link.springer.com/article/10.1007/s42994-022-00082-5</a>
</div>
<div id="ref-stuart2021single" class="csl-entry" role="listitem">
Stuart, T., Srivastava, A., Madad, S., Lareau, C. A., &amp; Satija, R. (2021). Single-cell chromatin state analysis with signac. <em>Nature Methods</em>, <em>18</em>(11), 1333â€“1341. <a href="https://www.nature.com/articles/s41592-021-01282-5">https://www.nature.com/articles/s41592-021-01282-5</a>
</div>
<div id="ref-teo2024best" class="csl-entry" role="listitem">
Teo, A. Y. Y., Squair, J. W., Courtine, G., &amp; Skinnider, M. A. (2024). Best practices for differential accessibility analysis in single-cell epigenomics. <em>Nature Communications</em>, <em>15</em>(1), 8805. <a href="https://www.nature.com/articles/s41467-024-53089-5">https://www.nature.com/articles/s41467-024-53089-5</a>
</div>
<div id="ref-wang2022benchmarking" class="csl-entry" role="listitem">
Wang, Y., Sun, X., &amp; Zhao, H. (2022). Benchmarking automated cell type annotation tools for single-cell ATAC-seq data. <em>Frontiers in Genetics</em>, <em>13</em>, 1063233. <a href="https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2022.1063233/full">https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2022.1063233/full</a>
</div>
<div id="ref-yan2020reads" class="csl-entry" role="listitem">
Yan, F., Powell, D. R., Curtis, D. J., &amp; Wong, N. C. (2020). From reads to insight: A hitchhikerâ€™s guide to ATAC-seq data analysis. <em>Genome Biology</em>, <em>21</em>, 1â€“16. <a href="https://link.springer.com/article/10.1186/s13059-020-1929-3">https://link.springer.com/article/10.1186/s13059-020-1929-3</a>
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/royfrancis\.github\.io\/seqhub");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../chapters/airr/index.html" class="pagination-link" aria-label="AIRR-Seq">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">AIRR-Seq</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../chapters/meta/index.html" class="pagination-link" aria-label="Meta transcriptomics">
        <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Meta transcriptomics</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>2025 Roy Francis</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
<p>Published with <a href="https://quarto.org/">Quarto</a> v1.6.43
</p>
</div>
  </div>
</footer>
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




<script src="../../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>