---
title: Spatial
description: Gene expression with spatial information
---

## Overview

There are three major resolution scales in ST data:

- multi-cell
- single-cell
- sub-cellular

@walker2022deciphering, @yue2023guidebook, @moses2022museum

## Techniques
### [10X Visium](https://www.10xgenomics.com/platforms/visium)

As of May 2024, there are three assays: Spatial GEX, Spatial GEX + Protein and Spatial GEX HD.

|Assay|Spatial GEX|Spatial GEX+Protein|Spatial GEX HD|
|---|---|---|---|
|Scope|Whole transcriptome|Whole transcriptome+31 Proteins|Whole transcriptome|
|Spatial resolution|multi-cell|multi-cell|single-cell|
|Tissue|FFPE,Fresh frozen,Fixed frozen|FFPE|FFPE|
|Species|Various|Human,Mouse|Human,Mouse|
|Capture area|6.5x6.5mm|6.5x6.5mm|6.5x6.5mm|
|Total spots|~5K/~14K|~5K/~14K|~11.2M|
|Spot size|55um circle|55um circle|2um square|

In addition, the Visium slides come in different types depending on the assay and version.

![Visium slide types. From left. Visium GEX V1 slide with 4 capture areas for direct use. Visium GEX V4 with 2 capture areas to be used with CytAssist. Visium GEX V5 with 2 larger capture areas to be used with CytAssist. Visium HD GEX H1 slide with 2 capture areas.](https://cdn.10xgenomics.com/image/upload/v1708031738/software-support/Spatial-GEX/SR-v3.0/SlideDiagrams-SerialNumbers.png)

![Visium HD GEX slide.](https://cdn.10xgenomics.com/image/upload/v1708031738/software-support/Spatial-GEX/SR-v3.0/SlideDefinitions.png)

- [Q&A](https://kb.10xgenomics.com/hc/en-us/categories/360002304851-Visium)
- [Spatial resolution and capture area](https://kb.10xgenomics.com/hc/en-us/articles/360035487572-What-is-the-spatial-resolution-and-configuration-of-the-capture-area-of-the-Visium-Gene-Expression-Slide-)
- [Visium glossary](https://www.10xgenomics.com/support/software/space-ranger/latest/getting-started/space-ranger-glossary)

## Methods/Tools

### Frameworks

- [{{< fa toolbox >}} Seurat (R)](https://satijalab.org/seurat/)
- [{{< fa toolbox >}} Giotto (R/Py)](https://rubd.github.io/Giotto_site/)
- [{{< fa toolbox >}} Squidpy (Py)](https://github.com/scverse/squidpy)
- [{{< fa toolbox >}} stLearn (Py)](https://stlearn.readthedocs.io/en/latest/)
- [{{< fa toolbox >}} Semla](https://ludvigla.github.io/semla/)

### SVG

Identification of spatially variable genes.

- {{< fa toolbox >}} SpatialDE
- {{< fa toolbox >}} SPARK
- {{< fa toolbox >}} SOMDE
- {{< fa toolbox >}} Sepal
- {{< fa toolbox >}} scGCO
- {{< fa toolbox >}} SpaGCN
- {{< fa toolbox >}} SpatialLIBD
- {{< fa toolbox >}} stLearn

### Spatial deconvolution

- [{{< fa toolbox >}} STdeconvolve (R)](https://github.com/JEFworks-Lab/STdeconvolve)
  - Reference-free

- [{{< fa toolbox >}} RCTD SpaceXR (R)](https://github.com/dmcable/spacexr)
  - Needs reference
  - Runs into error

- [{{< fa toolbox >}} SPOTlight (R)](https://github.com/MarcElosua/SPOTlight)
  - Uses reference

- [{{< fa toolbox >}} Giotto SpatialDWLS (R)](https://github.com/drieslab/Giotto)
- [{{< fa toolbox >}} SpatialDDLS (R)](https://github.com/diegommcc/SpatialDDLS)
- [{{< fa toolbox >}} DSTG (Py)](https://github.com/Su-informatics-lab/DSTG)
- [{{< fa toolbox >}} Tangram (Py)](https://github.com/broadinstitute/Tangram)

### Cell interaction

Given the relative stability of cellular locations, spatial transcriptomics allows us to reveal cellâ€“cell interactions (CCI),
also referred to as cell-cell communications (CCC), with fewer false positives than similar analysis with scRNA-seq data.

- {{< fa toolbox >}} SpaOTsc

## Datasets

- R package [TENxVisiumData](https://bioconductor.org/packages/release/data/experiment/vignettes/TENxVisiumData/inst/doc/vignette.html)

## Resources

- [OSTA Book](https://lmweber.org/OSTA-book/)
- [Elixir Spatial Omics Data Analysis Workshop](https://elixir-europe-training.github.io/ELIXIR-SCO-spatial-omics/schedule.html)

## References {.unnumbered}
